### PR TITLE
Fix errors found while making tests

### PR DESCRIPTION
**Pull request by jessberg**
I'm making tests for the output of the simulator, and this is just a hodgepodge of errors I found while making tests.  The main ones are:
1. Changing the edge dequeue headers to only the ones that are actually used
2. Changing how we deal with properties headers and path headers in graph_utils - we do not want to split by whitespace, but rather by the comma dividing different nodes.
3. Changing how node is displayed so that ID always comes first (makes it easier to trace the output)
4. Fix a bug in the simulator wherein the simulator forgot to consume the rpcs in the rpc buffer once it had passed them on.  Before, the rpcs were before staying in the rpc buffer when they should have been consumed, which resulted in duplicate rpcs.

### CODE REVIEW

File: libs/graph_utils/src/graph_utils.rs [hunk 1]
```diff
@@ -70,10 +70,10 @@ pub fn generate_trace_graph_from_headers(
     properties_header: String,
 ) -> Graph<(String, HashMap<String, String>), String> {
     let mut graph = Graph::new();
-    let mut nodes_iterator = paths_header.split_whitespace();
+    let mut nodes_iterator = paths_header.split(",");
     let mut node_str_to_node_handle = HashMap::new();
     let first_node = nodes_iterator.next();
-    if first_node.is_none() {
+    if first_node.is_none() || first_node.unwrap().trim().is_empty() {
```
Comment (fruffy) [inline libs/graph_utils/src/graph_utils.rs:76]: Why is this needed?

File: libs/graph_utils/src/graph_utils.rs [hunk 2]
```diff
@@ -114,15 +114,15 @@ pub fn generate_trace_graph_from_headers(
...
+                let node = period_iterator.next().unwrap().to_string();
                 let mut property = String::new();
                 for quality in period_iterator {
                     property.push_str(&quality);
                     property.push_str(".");
                 }
                 let _ = property.pop(); // get rid of trailing period
                 let node_weight: &mut (String, HashMap<String, String>) = graph
-                    .node_weight_mut(node_str_to_node_handle[node])
+                    .node_weight_mut(node_str_to_node_handle[&node])
```
Comment (fruffy) [inline libs/graph_utils/src/graph_utils.rs:125]: I do not think this is needed if you remove the to_string() call.

File: sim/src/simulator.rs [hunk 3]
```diff
@@ -111,7 +111,9 @@ impl Simulator {
         // tick all elements to generate Rpcs
         for (i, element) in self.elements.iter_mut() {
             let rpcs = element.tick(tick);
-            self.rpc_buffer.insert(i.to_string(), rpcs);
```
Comment (fruffy) [inline sim/src/simulator.rs:114]: The entire tick function is in really bad shape. We should clean it up sometime.

File: sim/src/simulator.rs [hunk 4]
```diff
@@ -131,15 +133,15 @@ impl Simulator {
         }
         for i in 0..self.elements.keys().count() {
             let src = &indices_to_sim_el[&i];
-            for (rpc, dst) in &self.rpc_buffer[src] {
+            while !&self.rpc_buffer[src].is_empty() {
+                let (rpc, dst) = &self.rpc_buffer.get_mut(src).unwrap().pop().unwrap();
```
Comment (fruffy) [inline sim/src/simulator.rs:138]: This a pretty complex unwrapping and accessing operation. Is this the right way to do this?

**Review by fruffy — CHANGES_REQUESTED** [review]
Great. It is good that these issue are now uncovered with the tests.

File: libs/graph_utils/src/graph_utils.rs (hunk 1 above)
Comment (jessberg) [reply libs/graph_utils/src/graph_utils.rs:76]: The first plugin to receive an RPC will get an empty paths header because the RPC hasn't visited anything yet.  This is checking if it is empty, because if so, we can just return.

File: sim/src/simulator.rs (hunk 4 above)
Comment (jessberg) [reply sim/src/simulator.rs:138]: I believe so.  The rpc buffer is a hashmap from the ids to a vector of rpcs, destinations.  So for each element, we are emptying the element's rpc buffer, and sending those rpcs to the corresponding destinations.  We have to do get_mut because we need to be able to delete rpcs from the buffer, and we have to do pop because we want to consume the rpcs.  Both those functions come with options, so it winds up being long for the unwraps.

File: sim/src/simulator.rs (hunk 3 above)
Comment (jessberg) [reply sim/src/simulator.rs:114]: I'll put a TODO

File: libs/graph_utils/src/graph_utils.rs (hunk 2 above)
Comment (jessberg) [reply libs/graph_utils/src/graph_utils.rs:125]: That's true;  it's fixed now.

File: libs/graph_utils/src/graph_utils.rs (hunk 1 above)
Comment (fruffy) [reply libs/graph_utils/src/graph_utils.rs:76]: Why is there a header if it is empty? Should it exist at that point?

File: sim/src/simulator.rs (hunk 4 above)
Comment (fruffy) [reply sim/src/simulator.rs:138]: Okay, I will take a closer look at this later.

**Review by fruffy — APPROVED** [review]
